From 3bd37b2a307b3f9fc87f34a433a810f2fc1c1760 Mon Sep 17 00:00:00 2001
From: Mao Minkai <maominkai@wxiat.com>
Date: Tue, 7 Jun 2022 17:42:04 +0800
Subject: [PATCH] sw64: add missing pkey syscall numbers
Patch-mainline: NO, OTHERS
References: openEuler-22.03-LTS
Modified-by-SEL: No


Sunway inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I56U83

--------------------------------

Most architectures define system call numbers for the pkey system calls
even when they don't support the features, and perhaps never will. sw64
miss these before, so just define them anyway for consistency.

Signed-off-by: Mao Minkai <maominkai@wxiat.com>

Signed-off-by: Gu Zitao <guzitao@wxiat.com>
Acked-by: Xie XiuQi <xiexiuqi@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/sw_64/include/uapi/asm/unistd.h   | 5 -----
 arch/sw_64/kernel/syscalls/syscall.tbl | 6 +++---
 2 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/arch/sw_64/include/uapi/asm/unistd.h b/arch/sw_64/include/uapi/asm/unistd.h
index 225358536dc9..c4fdde96c640 100644
--- a/arch/sw_64/include/uapi/asm/unistd.h
+++ b/arch/sw_64/include/uapi/asm/unistd.h
@@ -9,9 +9,4 @@
 
 #include <asm/unistd_64.h>
 
-/* sw64 doesn't have protection keys. */
-#define __IGNORE_pkey_mprotect
-#define __IGNORE_pkey_alloc
-#define __IGNORE_pkey_free
-
 #endif /* _UAPI_ASM_SW64_UNISTD_H */
diff --git a/arch/sw_64/kernel/syscalls/syscall.tbl b/arch/sw_64/kernel/syscalls/syscall.tbl
index 37a46ec99ccf..6026c637d923 100644
--- a/arch/sw_64/kernel/syscalls/syscall.tbl
+++ b/arch/sw_64/kernel/syscalls/syscall.tbl
@@ -295,9 +295,9 @@
 285	common	pidfd_getfd			sys_pidfd_getfd
 286	common	faccessat2			sys_faccessat2
 287	common	process_madvise			sys_process_madvise
-#288 is unused
-#289 is unused
-#290 is unused
+288	common	pkey_mprotect			sys_pkey_mprotect
+289	common	pkey_alloc			sys_pkey_alloc
+290	common	pkey_free			sys_pkey_free
 #291 is unused
 #292 is unused
 #293 is unused
-- 
2.34.1

